File: example-java/app/src/main/java/com/vibes/androidsdkexampleapp/api/VibesAPIContract.java
package com.vibes.androidsdkexampleapp.api;

import com.vibes.vibes.Credential;
import com.vibes.vibes.VibesListener;

/**
 * Created by jean-michel.barbieri on 2/25/18
 * Copyright (c) Vibes 2018 . All rights reserved.
 * Last modified 12:33 AM
 */
public interface VibesAPIContract {
    void registerDevice(VibesListener<Credential> listener);
    void unregisterDevice(VibesListener<Void> listener);
    void registerPush(VibesListener<Void> listener, String firebasePushToken);
    void unregisterPush(VibesListener<Void> listener);
}

File: example-kotlin/app/src/main/java/com/android/exampleapp/kotlin/api/VibesAPIContract.kt
package com.android.exampleapp.kotlin.api

import com.vibes.vibes.Credential
import com.vibes.vibes.VibesListener

interface VibesAPIContract {
    fun registerDevice(listener: VibesListener<Credential>)
    fun unregisterDevice(listener: VibesListener<Void?>)
    fun registerPush(listener: VibesListener<Void?>, firebasePushToken: String)
    fun unregisterPush(listener: VibesListener<Void?>)
}
